File: 1524-number-of-sub-arrays-with-odd-sum/1524-number-of-sub-arrays-with-odd-sum.py
class Solution:
    def numOfSubarrays(self, arr: List[int]) -> int:
        count = [1, 0]
        sumt = 0
        ans = 0 
        for i in arr:
            sumt+=i
            if sumt&1:
                ans += count[0]
                count[1]+=1
            else:
                ans += count[1]
                count[0]+=1
        return ans%(1000000007)

File: 1524-number-of-sub-arrays-with-odd-sum/1524-number-of-sub-arrays-with-odd-sum.java
class Solution {
    public int numOfSubarrays(int[] arr) {
        int even=1, odd=0;
        long sumt=0, ans=0;
        for(int a: arr){
            sumt+=(long)a;
            if(sumt%2==0){
                ans += (long)odd;
                even++;
            }else{
                ans += (long)even;
                odd++;
            }
        }
        return (int)(ans%(1000000007));
    }
}